force-app/main/default/classes/ContactTriggerExpensesService.cls

```diff
@@ -0,0 +1,34 @@
+public with sharing class ContactTriggerExpensesService {
+    public static List<Account> CheckingIsTheirExpensesChange(List<Contact> newTrigger,Map<ID, Contact> oldTrigger,Map<ID, Account> allAccounts){
+         System.debug('CheckingIsTheirExpensesChange');
+        List<Contact> contactForNextStep=new List<Contact>();
+        for(Contact contactNew:newTrigger){
+            if(contactNew.Expenses__c!=oldTrigger.get(contactNew.id).Expenses__c){
+                System.debug('CheckingIsTheirExpensesChange if reached');
+                contactForNextStep.add(contactNew);
+            }
+        }
+        return preparingAccountsForUpdateAfterExpensesChange(allAccounts, contactForNextStep);
+    
+    }
+    
+    
+    public static List<Account> preparingAccountsForUpdateAfterExpensesChange(Map<ID, Account> allAccounts,List<Contact> contactForNextStep){
+        System.debug('preparingAccountsForUpdate');
+        List<Account> accountsForUpdate=new List<Account>();
+        for(Contact contact:contactForNextStep){
+                Account account=allAccounts.get(contact.AccountId);
+                if(account.NumberOfBoughtCars__c==0){
+                        System.debug('Number of BoughtCars is 0 before trigger');
+                        account.Wallet__c=account.Budget__c;
+                    }
+                System.debug('preparingAccountsForUpdate s for reached');
+                account.Wallet__c-=contact.LastAddedExpenseValue__c;
```

Comment:
Tutaj takie podejście nigdy nie będzie działać na dużą skale, wystarczy, że ktoś usunie naraz 3 auta, to wtedy Ci lastAdded zostanie jako to najnowsze auto i kaplica. Spróbuj tutaj w ciele tej metody zaimplementować taki silnik jak napisałęm w komentarzu do liczenia kosztów na Contactie podczas edycji Carów. I ogranicz się tylko do liczenia expensy czyli musisz zebrać Id accountów z CONTACTÓW które wpadły w trigger, zapytać baze o Accounty i ich Contacty w jednej querzy po relacji tak jak zrobiłem to wcześniej z Contactami i Carami w komentarzu i wtedy sie przepętlić i obliczyć to dodająć expense i uzupełnić pole wtedy dopiero, pod tytułem Wallet, które będzie budżet - kwota która wyjdzie Ci z obliczeń